Source/SwiftLintBuiltInRules/Rules/RuleConfigurations/IndentationStyleConfiguration.swift

```diff
@@ -0,0 +1,61 @@
+import SwiftLintCore
+
+struct IndentationStyleConfiguration: SeverityBasedRuleConfiguration, Equatable {
+	typealias Parent = IndentationStyleRule
+
+	@ConfigurationElement(key: "severity")
+	private(set) var severityConfiguration = SeverityConfiguration<Parent>.warning
+	@ConfigurationElement(key: "include_multiline_strings")
+	private(set) var includeMultilineStrings = false
+	@ConfigurationElement(key: "include_multiline_comments")
+	private(set) var includeMultilineComments = true
+	@ConfigurationElement(key: "per_file")
+	private(set) var perFile = true
+	@ConfigurationElement(key: "preferred_style")
+	private(set) var preferredStyle = PreferredStyle.spaces
+	@ConfigurationElement(key: "tab_width")
+	private(set) var tabWidth: Int?
```

Comment:
Maybe it would be better with a different name. This isn't for setting how many spaces a tab should consist of, it's to tell swiftlint how wide your tabs are, since I don't think that can be derived from anywhere...

The reason being, if you are using tab indentation, sometimes you use spaces to provide a little nudging after the tab indentation. This setting allows for no more than `tabWidth - 1` spaces following any tab indentation.

I'll point out that, if you are using tabs in what I consider to be the most optimal way, you don't need this. However, forcing tabs projects to use tabs *my* way goes entirely against the philosophy of tabs usage in the first place, so this setting now exists.

Reply:
This would also be required, if the we offer an automatic rewrite for the rule.

But I wonder, what happens if it remains `nil`. Then the rule would have to assume something, wouldn't it? Is it better to specify 2 or 4 as defaults right away then?